### DISCUSSION TITLE
When changing models mid conversation does LibreChat send the whole conversation history to the newly selected model?

### DISCUSSION THREAD

**Discussion by anose001** [danny-avila/LibreChat#3999] [Q&A] [answered]
This comment [here](https://community.openai.com/t/conversation-context-and-quadratic-billing/126421#:~:text=I%20think%20you,it%20works%2C%20yes.) suggest that all the histroy of a conversation counts towards the token count. This means that the cost of the next message send in a conversation increases non-linearly with each additional message. This can spiral out of hand for really long messages.

@danny-avila, in this context, would a Token Counter be helpful?

**Comment by danny-avila** [comment, marked as answer]
Of course, and there's many ways you can mitigate this. For example, you can utilize a preset to keep the context window intentionally at a limit you are comfortable with; otherwise, it will use the known limit for the model you have selected. For gpt-4o, this is a max of 128K tokens

![image](https://github.com/user-attachments/assets/97164e7e-6ead-427f-b847-3f0a441aff15)

**Reply by danny-avila** [reply]
Also forking: https://www.librechat.ai/docs/user_guides/fork

**Reply by BIG-debug-admin** [reply]
Hey @anose001 and @danny-avila  - I actually created a solution to the exact issue you are describing. Inkfold is one workspace for ChatGPT, Claude, and Gemini with a memory layer that belongs to you, not to any provider. Switch models mid-conversation and your context comes with you, so you stop re-explaining yourself every time you change tools. Right now, we are looking for people to try the product and see if there are any bugs. To get Inkfold for free: https://app.inkfold.app/tester-signup. Would love to further connect!
